src/acl.c

```diff
@@ -638,6 +638,43 @@ void ACLSetSelectorCommandBitsForCategory(dict *commands, aclSelector *selector,
     dictReleaseIterator(di);
 }
 
+/* This function is responsible for recomputing the command bits for all selectors of the existing users.
+ * It uses the 'command_rules', a string representation of the ordered categories and commands, 
+ * to recompute the command bits. */
+void ACLRecomputeCommandBitsFromCommandRulesAllUsers() {
+    raxIterator ri;
+    raxStart(&ri,Users);
+    raxSeek(&ri,"^",NULL,0);
+    while(raxNext(&ri)) {
+        user *u = ri.data;
+        listIter li;
+        listNode *ln;
+        listRewind(u->selectors,&li);
+        while((ln = listNext(&li))) {
+            aclSelector *selector = (aclSelector *) listNodeValue(ln);
+            int argc = 0;
+            sds *argv = sdssplitargs(selector->command_rules, &argc);
+            serverAssert(argv != NULL);
+            if (ACLSelectorCanExecuteFutureCommands(selector)) {
+                int res = ACLSetSelector(selector,"+@all",-1);
+                serverAssert(res == C_OK);
+            } else {
+                int res = ACLSetSelector(selector,"-@all",-1);
+                serverAssert(res == C_OK);
+            }
```

Comment:
> btw, why not use selector->flags & SELECTOR_FLAG_ALLCOMMANDS instead of ACLSelectorCanExecuteFutureCommands? or am i missing something?

They do the same thing. I assume part of this is party with ACLDescribeSelectorCommandRules which does the same thing as part of the "sanity" check code. We could probably simplify the code.

Reply:
> let's add a comment for this block saying that it's done to clean the slate. (assuming i'm right)

yes, you are right!

> > btw, why not use selector->flags & SELECTOR_FLAG_ALLCOMMANDS instead of ACLSelectorCanExecuteFutureCommands? or am i missing something?
> 
> They do the same thing. I assume part of this is parity with ACLDescribeSelectorCommandRules which does the same thing as part of the "sanity" check code. We could probably simplify the code.

I will look into it!

Reply:
> I will look into it!

Just going to say this is not a high priority, let's consider refactoring this later.